### DISCUSSION TITLE
Re: Required Components - There's a huge step backwards in usability that may have been overlooked...

### DISCUSSION THREAD

**Discussion by Veliladon** [bevyengine/bevy#16570] [General]
Ok so I'll admit from the outset I'm mostly on Discord. I don't follow Github issues or discussions about these architectural decisions so it's possible I missed a whole big conversation over this. I looked for discussions on required components and deprecating bundles but my search skills are insufficient. Anyway, this is Bevy 0.15's required components vs bundles:

![bundlevsreqcomp](https://github.com/user-attachments/assets/e1a718de-d5b5-4ec1-b005-2d3cb1ec5bc0)

The bundle I can see what I can customize just by mousing over the struct. The SceneRoot will insert Transform and Visibility components but how do I know what I can customize without having to go look up the component in docs and check what gets inserted as a required component? I don't even know what I'm about to insert without looking at docs. Instead of being able to look at a bundle and go "ooh, that's probably what I'm looking for" I have to go back to docs to find the component by hand each and every time.

Like I'm sure there's a better workflow I can use and I'm willing to adapt if this is the definitive way forward and I'm more than willing to use other people's ideas for adapting to it. But in its current state I'm just not vibing with it and could use some help and/or guidance.

**Comment by Veliladon** [comment]
So Bruno on Discord pointed out ctrl-click on the struct name takes you to the source where the required components are listed. It's workable at least compared to switching out to a docs.rs Chrome tab. Anyone with any other workflow tips or suggestions for surviving the change would be welcome.

**Comment by aloucks** [comment]
I've run into this issue as well. The docs are generated on the implementation for `Component` rather than on the struct itself and the tooltip doesn't show the docs for the impls. Additionally, even when browsing the source, the conditional require on Visibility doesn't work with goto-def. 

Is there any opposition to rendering the component docs on the struct itself? Maybe in addition to on the Component impl?
